services/f-droid/f-droid.service.js

```diff
@@ -1,19 +1,28 @@
 'use strict'
 
 const Joi = require('@hapi/joi')
+const {
+  optionalNonNegativeInteger,
+  nonNegativeInteger,
+} = require('../validators')
 const { addv } = require('../text-formatters')
 const { version: versionColor } = require('../color-formatters')
-const { BaseYamlService, InvalidResponse } = require('..')
+const { BaseJsonService, NotFound } = require('..')
 
 const schema = Joi.object({
-  CurrentVersion: Joi.alternatives().try(Joi.string(), Joi.number()).required(),
+  packageName: Joi.string().required(),
+  suggestedVersionCode: optionalNonNegativeInteger,
+  packages: Joi.array().items({
+    versionName: Joi.string().required(),
+    versionCode: nonNegativeInteger,
+  }),
 }).required()
 
 const queryParamSchema = Joi.object({
-  metadata_format: Joi.string().valid('yml', 'txt'),
+  include_prereleases: Joi.equal(''),
```

Comment:
Does this change of query params have the potential to impact the message that could be displayed on any existing user badges?

Reply:
In the worst case "package not found" error will be displayed for them. This might be the case for those who publish only beta versions into F-Droid and use the badge (I guess there no such cases).

I think the fact badge is moved from repo metadata to API can be worse, e.g. it won't work for archived applications (which doesn't have packages in the main repository), however they're also not available via F-Droid website.

Removing `metadata_format` shouldn't impact at all.

Reply:
I'm not sure I followed all that (I know nothing about F-Droid :smile:). IIUC, the query param change won't have any impact but there's a very small risk that the change of data source/API could break existing badges in a very rare set of circumstances, and we wouldn't anticipate any such existing badges.

That SGTM